### DISCUSSION TITLE
Best approach to unregister nested fields

### DISCUSSION THREAD

**Discussion by danieladarve** [react-hook-form/react-hook-form#3954] [Q&A] [answered]
Hello people,

Consider the following data structure for a repeater field that contains other fields in a dynamic form, where "new1, new2" are the rows containing the repeater fields. New rows can be added or removed using a button.
```
{
        otherFieldHandleName: "value",
        repeaterHandleName: {
          rows: {
            new1: {
              fields: {
                fieldHandle: "value",
                field2Handle: "value",
                field3Handle: "value",
              },
            },
            new2: {
              fields: {
                fieldHandle: "value",
                field2Handle: "value",
                field3Handle: "value",
              },
            },
            new3: {
              fields: {
                fieldHandle: "value",
                field2Handle: "value",
                field3Handle: "value",
              },
            },
          },
        },
}
```
I am wondering what would be the best approach to unregister nested fields ( or a full row ) when removing a row.

I have tried removing each field using a loop:
unregister(`${handle}.rows.new2.fields.${field.handle}`)
unregister(`${handle}['rows']['new2']['fields']['${field.handle}']`)

and removing the row itself:
unregister(`${handle}.rows.new${row}`)
unregister(`${handle}['rows']['new${row}']`)

not working for me, any ideas of what I am missing?

Cheers

**Comment by bluebill1049** [comment, marked as answer]
current v6 should unregister input automatically when they r removed. do you have a csb?

**Reply by danieladarve** [reply]
Hey @bluebill1049  thanks for getting back to me, you are correct, they do unregister automatically. It was a mistake on my end.

**Reply by danieladarve** [reply]
@bluebill1049 sorry to bother you again, one thing i just noticed is, while the fields do get unregistered, If I was to add 4 rows, and for instance I delete the 2nd row from top to bottom. while the field names are kept, the values do shift up.

So in this case, for: 
row1: values stay the same
row2: gets unregistered
row3: takes the values of the unregistered row2

Same thing happens, if you add 3 rows and delete the 1st row, you will see how the values start shifting up.

Any ideas? this is similar to the initial issue i was having at the start, and thats why i was thinking I needed to add logic to unregister the fields.

Thanks

**Reply by bluebill1049** [reply]
are you using `useFieldArray`?

**Reply by danieladarve** [reply]
no, I am using the register method passed down via the FormProvider context.

**Reply by bluebill1049** [reply]
sounds like it's better to use `useFieldArray`, when you delete and manipulate multiple fields.
